src/index/TextIndexBuilder.cpp

```diff
@@ -338,134 +338,29 @@ void TextIndexBuilder::createTextIndex(const string& filename, TextVec& vec) {
-
-  // If there is a four letter prefix aaaa, ...., zzzz in `forcedBlockStarts`
-  // the `SortKey` of which is a prefix of `prefixSortKey`, then set
-  // `prefixSortKey` to that `SortKey` and `prefixLength` to
-  // `MIN_WORD_PREFIX_SIZE` (4). This ensures that the blocks corresponding to
-  // these prefixes are never split up because of Unicode ligatures.
-  auto adjustPrefixSortKey = [&](auto& prefixSortKey, auto& prefixLength) {
-    while (true) {
-      if (forcedBlockStartsIt == forcedBlockStarts.end()) {
-        break;
-      }
-      auto forcedBlockStartSortKey = locManager.getSortKey(
-          *forcedBlockStartsIt, LocaleManager::Level::PRIMARY);
-      if (forcedBlockStartSortKey >= prefixSortKey) {
-        break;
-      }
-      if (prefixSortKey.starts_with(forcedBlockStartSortKey)) {
-        prefixSortKey = std::move(forcedBlockStartSortKey);
-        prefixLength = MIN_WORD_PREFIX_SIZE;
-        return;
-      }
-      forcedBlockStartsIt++;
-    }
-  };
-
-  auto getLengthAndPrefixSortKey = [&](WordVocabIndex i) {
-    auto word = index.textVocab_[i];
-    auto [len, prefixSortKey] =
-        locManager.getPrefixSortKey(word, MIN_WORD_PREFIX_SIZE);
-    if (len > MIN_WORD_PREFIX_SIZE) {
-      LOG(DEBUG) << "The prefix sort key for word \"" << word
-                 << "\" and prefix length " << MIN_WORD_PREFIX_SIZE
-                 << " actually refers to a prefix of size " << len << '\n';
-    }
-    // If we are in a block where one of the fourLetterPrefixes are contained,
-    // use those as the block start.
-    adjustPrefixSortKey(prefixSortKey, len);
-    return std::tuple{std::move(len), std::move(prefixSortKey)};
-  };
-  auto [currentLen, prefixSortKey] =
-      getLengthAndPrefixSortKey(WordVocabIndex::make(0));
-  for (size_t i = 0; i < index.textVocab_.size() - 1; ++i) {
-    // we need foo.value().get() because the vocab returns
-    // a std::optional<std::reference_wrapper<string>> and the "." currently
-    // doesn't implicitly convert to a true reference (unlike function calls)
-    const auto& [nextLen, nextPrefixSortKey] =
-        getLengthAndPrefixSortKey(WordVocabIndex::make(i + 1));
-
-    bool tooShortButNotEqual =
-        (currentLen < MIN_WORD_PREFIX_SIZE || nextLen < MIN_WORD_PREFIX_SIZE) &&
-        (prefixSortKey != nextPrefixSortKey);
-    // The `startsWith` also correctly handles the case where
-    // `nextPrefixSortKey` is "longer" than `MIN_WORD_PREFIX_SIZE`, e.g. because
-    // of unicode ligatures.
-    bool samePrefix = nextPrefixSortKey.starts_with(prefixSortKey);
-    if (tooShortButNotEqual || !samePrefix) {
-      blockBoundaryAction(i);
-      numBlocks++;
-      currentLen = nextLen;
-      prefixSortKey = nextPrefixSortKey;
-    }
+  size_t numBlocks =
+      std::ceil(static_cast<float>(vocabSize) / static_cast<float>(blockSize));
+  for (size_t currentId = blockSize - 1; currentId < vocabSize - 1;
+       currentId += blockSize) {
+    blockBoundaryAction(currentId);
```

Comment:
Am I seeing this correctly, that currently the blocksize is `number of words in the block, doesn't matter how often a word appears in a block` , as this has the least impliciations on the existing code? 
What would happen if we were to say `100'000 word mentions are a block`? or is this a problem because of the way the word and entity mentions are stored, and thus the words have to stay together?